nixos/modules/services/misc/jellyfin.nix

```diff
@@ -52,9 +64,9 @@ in
       serviceConfig = rec {
         User = cfg.user;
         Group = cfg.group;
-        StateDirectory = "jellyfin";
-        CacheDirectory = "jellyfin";
-        ExecStart = "${cfg.package}/bin/jellyfin --datadir '/var/lib/${StateDirectory}' --cachedir '/var/cache/${CacheDirectory}'";
+        StateDirectory = cfg.dataDir;
+        CacheDirectory = cfg.cacheDir;
```

Comment:
these are invalid according to systemd.exec(5):
```
       RuntimeDirectory=, StateDirectory=, CacheDirectory=, LogsDirectory=, ConfigurationDirectory=
           These options take a whitespace-separated list of directory names. The specified directory
           names must be relative, and may not include "..".
```

Reply:
don't know enough about jellyfin to make an informed decision, but it feels like we should bind mount them instead if possible. systemd will create these directories if they don't exist, that's a good bit of convenience for the default setup and something we'd have to emulate in a preStart script if they went away.

Reply:
How do you suggest me to proceed in this case then @pennae?

Reply:
easiest way would be to hardcode both to `jellyfin` and have users who want to use different directories bind mount those by adding their own `BindPaths`. most users *probably*(?) won't want to change those anyway.